netutils/lib_mapper.py

```diff
@@ -641,3 +647,73 @@
 _MAIN_LIB_MAPPER["watchguard_firebox"] = "watchguard_firebox"
 _MAIN_LIB_MAPPER["windows"] = "windows"
 MAIN_LIB_MAPPER: t.Dict[str, str] = {key: _MAIN_LIB_MAPPER[key] for key in sorted(_MAIN_LIB_MAPPER)}
+
+NAME_TO_LIB_MAPPER: t.Dict[str, t.Dict[str, str]] = {
+    "aerleon": AERLEON_LIB_MAPPER,
+    "ansible": ANSIBLE_LIB_MAPPER,
+    "capirca": CAPIRCA_LIB_MAPPER,
+    "dna_center": DNACENTER_LIB_MAPPER,
+    "forward_networks": FORWARDNETWORKS_LIB_MAPPER,
+    "hier_config": HIERCONFIG_LIB_MAPPER,
+    "napalm": NAPALM_LIB_MAPPER,
+    "netmiko": NETMIKO_LIB_MAPPER,
+    "netutils_parser": NETUTILSPARSER_LIB_MAPPER,
+    "nist": NIST_LIB_MAPPER,
+    "ntc_templates": NTCTEMPLATES_LIB_MAPPER,
+    "pyats": PYATS_LIB_MAPPER,
+    "pyntc": PYNTC_LIB_MAPPER,
+    "scrapli": SCRAPLI_LIB_MAPPER,
+}
+
+
+NAME_TO_LIB_MAPPER_REVERSE: t.Dict[str, t.Dict[str, str]] = {
+    "aerleon": AERLEON_LIB_MAPPER_REVERSE,
+    "ansible": ANSIBLE_LIB_MAPPER_REVERSE,
+    "capirca": CAPIRCA_LIB_MAPPER_REVERSE,
+    "dna_center": DNACENTER_LIB_MAPPER_REVERSE,
+    "forward_networks": FORWARDNETWORKS_LIB_MAPPER_REVERSE,
+    "hier_config": HIERCONFIG_LIB_MAPPER_REVERSE,
+    "napalm": NAPALM_LIB_MAPPER_REVERSE,
+    "netmiko": NETMIKO_LIB_MAPPER_REVERSE,
+    "netutils_parser": NETUTILSPARSER_LIB_MAPPER_REVERSE,
+    "nist": NIST_LIB_MAPPER_REVERSE,
+    "ntc_templates": NTCTEMPLATES_LIB_MAPPER_REVERSE,
+    "pyats": PYATS_LIB_MAPPER_REVERSE,
+    "pyntc": PYNTC_LIB_MAPPER_REVERSE,
+    "scrapli": SCRAPLI_LIB_MAPPER_REVERSE,
+}
+
+
+def get_lib_mappings(
+    lib_mapping_reverse: t.Optional[t.Dict[str, t.Dict[str, str]]] = None
+) -> t.Dict[str, t.Dict[str, str]]:
+    """Generates a dictionary of all available network driver mappings.
+
+    Args:
+        lib_mapping_reverse: A reverse mapping of library names to their normalized names. Defaults to NAME_TO_LIB_MAPPER_REVERSE if not provided.
```

Comment:
What would be the context in which you'd provide a non-default `lib_mapping_reverse` value? I don't see any examples in this PR or in the tests.

Reply:
I was following a pattern that was similar to one described here: https://github.com/networktocode/netutils/issues/158#issuecomment-1287046776 where you could override certain variables, but let me think about it, perhaps it does not make sense.